dbt/include/bigquery/macros/materializations/incremental_strategy/insert_overwrite.sql

```diff
@@ -47,37 +47,63 @@
   {% endif %}
 {% endmacro %}
 
+{% macro bq_static_select_insert_overwrite_sql(tmp_relation, sql, partition_by, tmp_relation_exists) %}
+  {%- set source_sql -%}
+  (
+    {% if partition_by.time_ingestion_partitioning and tmp_relation_exists -%}
+    select
+      {{ partition_by.insertable_time_partitioning_field() }},
+      * from {{ tmp_relation }}
+    {% elif tmp_relation_exists -%}
+      select
+      * from {{ tmp_relation }}
+    {%- elif partition_by.time_ingestion_partitioning -%}
+      {{ wrap_with_time_ingestion_partitioning_sql(partition_by, sql, True) }}
+    {%- else -%}
+      {{sql}}
+    {%- endif %}
+  )
+  {%- endset -%}
+  {{ return(source_sql) }}
+{% endmacro %}
+
+{% macro bq_static_copy_partitions_insert_overwrite_sql(
+  tmp_relation, target_relation, sql, partition_by, partitions, tmp_relation_exists
+  ) %}
+  {%- if tmp_relation_exists is false -%}
+    {%- set source_sql = bq_static_select_insert_overwrite_sql(tmp_relation, sql, partition_by, tmp_relation_exists) %}
+  {# We run temp table creation in a separated script to move to partitions copy if it doesn't already exist #}
+    {%- call statement('create_tmp_relation_for_copy', language='sql') -%}
+      {{ bq_create_table_as(partition_by, True, tmp_relation, source_sql, 'sql')
+    }}
+    {%- endcall %}
+  {%- endif -%}
+  {%- set partitions_sql -%}
+    select
+    {%- for partition in partitions %}
+      CAST({{ partition }} AS TIMESTAMP){%- if not loop.last -%},{%- endif -%}
+    {%- endfor %}
+    from {{ tmp_relation }}
+  {%- endset -%}
+  {%- set partitions = run_query(partitions_sql).columns[0].values() -%}
+  {# We copy the partitions #}
+  {%- do bq_copy_partitions(tmp_relation, target_relation, partitions, partition_by) -%}
+  -- Clean up the temp table
+  drop table if exists {{ tmp_relation }}
+{% endmacro %}
+
 {% macro bq_static_insert_overwrite_sql(
     tmp_relation, target_relation, sql, partition_by, partitions, dest_columns, tmp_relation_exists, copy_partitions
 ) %}
-
+  {%- if copy_partitions %}
```

Comment:
Is this always the case that copy_partitions leads to the static ('by value') method?

Reply:
I'm not sure I got the question: `copy_partitions` option is available to deal with dynamic and static partitioning. So that it doesn't always lead to static partitioning methods.